### DISCUSSION TITLE
How does this work?

### DISCUSSION THREAD

**Discussion by test3211234** [helix-editor/helix#11128] [Q&A]
https://github.com/helix-editor/helix/blob/649bd4501ee3f292d57964b8a8504fd9a83bd6e8/helix-term/src/commands.rs#L94

I see this context is passed into each function, but I never see it actually get created. I tried GitHub searching but couldn't find anything.

**Comment by kirawi** [comment]
GitHub uses tree-sitter so it will never be 100% accurate.
https://github.com/helix-editor/helix/blob/649bd4501ee3f292d57964b8a8504fd9a83bd6e8/helix-term/src/ui/editor.rs#L1317-L1324

**Reply by test3211234** [reply]
Oh, ok. But I still don't get this: 

https://github.com/helix-editor/helix/blob/649bd4501ee3f292d57964b8a8504fd9a83bd6e8/helix-term/src/compositor.rs#L170

This shows a callback. Is this where the command is actually being called or in the function you linked? The different contexts confuse me.
